ex1/8.ts

```diff
@@ -0,0 +1,99 @@
+/*
+
+Intro:
+
+    Project grew and we ended up in a situation with
+    some users starting to have more influence.
+    Therefore, we decided to create a new person type
+    called PowerUser which is supposed to combine
+    everything User and Admin have.
+
+Exercise:
+
+    Define type PowerUser which should have all fields
+    from both User and Admin (except for type),
+    and also have type 'powerUser' without duplicating
+    all the fields in the code.
+
+*/
+
+interface User {
+    type: 'user';
+    name: string;
+    age: number;
+    occupation: string;
+}
+
+interface Admin {
+    type: 'admin';
+    name: string;
+    age: number;
+    role: string;
+}
+
+type PowerUser = {
+    type: 'powerUser';
+    name: string;
+    age: number;
+    role: string;
+    occupation: string;
+};
```

Comment:
```suggestion
type PowerUser = {
    type: 'powerUser';
};
```
chỗ này có cách nào k repeat lại code kiểu kiểu ntn k anh nhỉ, thử xem ạ